python/src/main/java/org/navgurukul/playground/editor/PythonEditorFragment.kt

```diff
@@ -118,7 +118,11 @@ class PythonEditorFragment : BaseFragment() {
                     if (sheetBehavior.state == BottomSheetBehavior.STATE_EXPANDED) {
                         // If bottom sheet is expanded, collapse it on back button
                         sheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
-                    } else {
+                    }
+                    if(!viewModel.viewState.value!!.fileSaved&&viewModel.viewState.value!!.code.isNotEmpty()) {
```

Comment:
viewModel.viewState.value!! could be saved in a variable and reused. Also lets pls add spaces around '&&' to maintain readability